lib/net/ntlm/encode_util.rb

```diff
@@ -27,7 +27,7 @@ def self.swap16(str)
       # Decode a UTF16 string to a ASCII string
       # @param [String] str The string to convert
       def self.decode_utf16le(str)
-        str.force_encoding(Encoding::UTF_16LE)
+        str.dup.force_encoding(Encoding::UTF_16LE)
```

Comment:
Don't we need to do an assignment here? Otherwise the next line will be against the original object?